### DISCUSSION TITLE
Overriding the function body while modifying the function's definition

### DISCUSSION THREAD

**Discussion by jakubtobiasz** [community/community#7698] [Copilot Conversations] [answered]
Hi!
I'm using the Copilot with PhpStorm and it's really great. The one thing which annoys me a lot is a situation when I have an already existing method/function and while modifying the method's definition the Copilot (e.g. adding a parameter) suggests the whole new method's body on the top of the existing one. I'm attaching a screen below to make it easier to understand :).

![image](https://user-images.githubusercontent.com/80641364/141692133-7af01b36-778f-4329-8e05-e5907e015fd0.png)

P.S. The code doesn't make any sense, it's just to visualize the problem ;).

**Comment by MulleDK19** [comment]
The problem here, really is the inability to specify whether you want the whole block or just the active line completed.

The reason it suggests the whole function again, is because it cannot see past the current caret position, so it doesn't know the function already has a body.

**Comment by jansorg** [comment, marked as answer]
Well, technically the plugin inside the IDE can find the end of the block after the cursor position and a different keyboard shortcut or action could replace the block instead of inserting the text.
I'll suggest this to the team, but can't promise anything at this time because this is difficult to get right for all supported languages.
